cpp/ql/src/semmle/code/cpp/ir/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -211,10 +211,17 @@ private predicate fieldStoreStepNoChi(Node node1, FieldContent f, PostUpdateNode
   )
 }
 
+private FieldAddressInstruction getFieldInstruction(Instruction instr) {
+  result = instr or
+  result = instr.(CopyValueInstruction).getUnary()
+}
```

Comment:
This predicate is repeated in two files. Ideally we'd have just one copy, and it would be in a more principled form that has less to do with the surface syntax.

What would be ideal? A member predicate `Instruction.getNonCopyDefinition` that skips backwards through all exact `CopyInstruction`s (not just `CopyValueInstruction`s)?

Reply:
`Instruction.getNonCopyDefinition` would be a useful thing to have, indeed! I think that's worth a separate PR, however.